content-services/content-sources-backend
# HMS-4934: Add modules to introspected data

## Summary

Adds module streams as searchable for introspected repos

## Testing steps

1.  Adjust go.mod to point to your localy checkout of yummy, which should have https://github.com/content-services/yummy/pull/31

```
replace github.com/content-services/yummy => /home/jlsherri/git/yummy/
```

2. `go get ./...`
3. ` make db-migrate-up`
4. `make repos-import`
5. ```go run cmd/external-repos/main.go introspect go run cmd/external-repos/main.go introspect https://cdn.redhat.com/content/dist/rhel8/8/x86_64/appstream/os```
6. ```go run cmd/external-repos/main.go introspect go run cmd/external-repos/main.go introspect https://cdn.redhat.com/content/dist/rhel9/9/x86_64/appstream/os```
7.   play around with the module streams search api:
```
#### module streams
POST http://localhost:8000/api/content-sources/v1.0/module_streams/search
x-rh-identity: eyJpZGVudGl0eSI6eyJ0eXBlIjoiVXNlciIsInVzZXIiOnsidXNlcm5hbWUiOiIxIn0sImFjY291bnRfbnVtYmVyIjoiZm9vIiwiaW50ZXJuYWwiOnsib3JnX2lkIjoiMSJ9fX0K
Content-Type: application/json

{"urls":["https://cdn.redhat.com/content/dist/rhel8/8/x86_64/appstream/os"],
  "search":"",
  "rpm_names": []
}
```

feel free to play with the search and rpm_names
